docs/rules/no-implicit-globals.md

```diff
@@ -0,0 +1,40 @@
+# disallow `var` and named functions in global scope (no-implicit-globals)
+
+When working with browser scripts, developers often forget that variable and function definitions at the top-level scope become global variables on the `window` object. As opposed to modules which have their own scope. Globals should be explicitly assigned to `window` or `self` if that is the intent. Otherwise variables intended to be local to the script should be wrapped in an IIFE.
+
+## Rule Details
```

Comment:
We need some description under this heading about how this rule won't warn in ES6 or CommonJS modules.